plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueExpressionUtil.java

```diff
@@ -147,6 +147,14 @@ static Optional<String> buildGlueExpressionForSingleDomain(String columnName, Do
             return Optional.empty();
         }
 
+        // Glue throws an exception (e.g. input string: "__HIVE_D" is not an integer)
+        // for column <> '__HIVE_DEFAULT_PARTITION__' or column = '__HIVE_DEFAULT_PARTITION__' expression on numeric types
+        // "IS NULL" operator in the official documentation always returns empty result regardless of the type.
+        // https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-partitions.html#aws-glue-api-catalog-partitions-GetPartitions
+        if ((domain.getValues().isAll() || domain.getValues().isNone()) && !isQuotedType(domain.getType())) {
+            return Optional.empty();
+        }
```

Comment:
The code below doesn't handle the `IS NULL OR IS some value` case correctly.
-- It's apparent from the fact it "recognizes" `IS NULL` case by inspecting `valueSet.isNone()` (which is quite implicit), instead of checking `domain.isNullAllowed()`. Thus, it doesn't discern "some values" and "some values OR NULL" cases.

https://github.com/trinodb/trino/pull/13214 should make it clear (it doesn't solve the problem just yet)

Reply:
Thanks for your explanation. I was going to handle `IS NULL OR IS some value` case in #13122 after this PR.